aptos-move/framework/supra-framework/sources/multisig_voting.move

```diff
@@ -0,0 +1,1219 @@
+        execution_content: Option<ProposalType>,
+
+        /// Optional. Value is serialized value of an attribute.
+        /// Currently, we have three attributes that are used by the voting flow.
+        /// 1. RESOLVABLE_TIME_METADATA_KEY: this is uesed to record the resolvable time to ensure that resolution has to be done non-atomically.
+        /// 2. IS_MULTI_STEP_PROPOSAL_KEY: this is used to track if a proposal is single-step or multi-step.
+        /// 3. IS_MULTI_STEP_PROPOSAL_IN_EXECUTION_KEY: this attribute only applies to multi-step proposals. A single-step proposal will not have
+        /// this field in its metadata map. The value is used to indicate if a multi-step proposal is in execution. If yes, we will disable further
+        /// voting for this multi-step proposal.
+        metadata: SimpleMap<String, vector<u8>>,
+
+        /// Timestamp when the proposal was created.
+        creation_time_secs: u64,
+
+        /// Required. The hash for the execution script module. Only the same exact script module can resolve this
+        /// proposal.
+        execution_hash: vector<u8>,
+
+        /// A proposal is only resolved if expiration has passed and the number of votes is above threshold.
+        min_vote_threshold: u128,
+        expiration_secs: u64,
+
+
+        /// Number of votes for each outcome.
+        /// u128 since the voting power is already u64 and can add up to more than u64 can hold.
+        yes_votes: u128,
+        no_votes: u128,
+
+        /// Whether the proposal has been resolved.
+        is_resolved: bool,
+        /// Resolution timestamp if the proposal has been resolved. 0 otherwise.
+        resolution_time_secs: u64,
+    }
+
+    struct VotingForum<ProposalType: store> has key {
+        /// Use Table for execution optimization instead of Vector for gas cost since Vector is read entirely into memory
+        /// during execution while only relevant Table entries are.
+        proposals: Table<u64, Proposal<ProposalType>>,
+        events: VotingEvents,
+        /// Unique identifier for a proposal. This allows for 2 * 10**19 proposals.
+        next_proposal_id: u64,
+    }
+
+    struct VotingEvents has store {
+        create_proposal_events: EventHandle<CreateProposalEvent>,
+        register_forum_events: EventHandle<RegisterForumEvent>,
+        resolve_proposal_events: EventHandle<ResolveProposal>,
+        vote_events: EventHandle<VoteEvent>,
+    }
+
+    #[event]
+    struct CreateProposal has drop, store {
+        proposal_id: u64,
+        execution_hash: vector<u8>,
+        expiration_secs: u64,
+        metadata: SimpleMap<String, vector<u8>>,
+        min_vote_threshold: u128,
+    }
+
+    #[event]
+    struct RegisterForum has drop, store {
+        hosting_account: address,
+        proposal_type_info: TypeInfo,
+    }
+
+    #[event]
+    struct Vote has drop, store {
+        proposal_id: u64
+    }
```

Comment:
In the original `Voting` does it also record in an event if the vote is a `yes` vote or a `no` vote? Just `proposal_id` is not useful.

Reply:
In the original voting module **Vote** has only two filed `proposal_id` and `num_votes`, there is no any record for **yes vote** or a **no vote**. I think it's better we can add it in our module

Reply:
I have added `is_vote_yes` field in the event